handlers/_swagger.js

```diff
@@ -259,48 +260,46 @@ function registerRoute(app, auth, additionalMiddleware, method, path, data, allo
+                                if (Array.isArray(body)) {
+                                    if (body.length === 0) {
+                                        body.push(validationErrors);
+                                    } else {
+                                        body[0]._response_validation_errors = validationErrors;
+                                        if (body.length > 1) {
+                                            body[body.length - 1]._response_validation_errors = validationErrors;
+                                        }
                                     }
+                                } else {
+                                    body = body || {};
+                                    body._response_validation_errors = _.clone(validationErrors);
                                 }
-                            } else {
-                                body = body || {};
-                                body._response_validation_errors = validationErrors;
+                            }
+                            else {//level is fail
+                                body = {response: body, validationErrors: _.clone(validationErrors)};
+                                res.statusCode = validationErrors.status;
                             }
                         }
-                    }
-                    
-                    //after this initial call (sometimes `send` will call itself again), we don't need to get the response for validation anymore
-                    res.send = responseSender;
-                    responseSender.call(res, isJson ? JSON.stringify(body): body);
-                    
-                    if (responseModelValidationLevel === 'warn') {
-                        //when doing a warning only, do the work after the response has already been sent
-                        validationErrors = validateResponse();
-                    }
-                    
-                    if (validationErrors) { // for both errors and warnings ...
                         validationErrors.invalidResponse = {
                             method: req.method,
                             path: req.path,
                             statusCode: res.statusCode,
                             body: invalidBody || body
                         };
-                        logger[responseModelValidationLevel]('Response validation error:', JSON.stringify(validationErrors, null, 2));
+                        logger[responseModelValidationLevel === 'warn' ? 'warn' : 'error']('Response validation error:', JSON.stringify(validationErrors, null, 2));
                     }
+
+                    //after this initial call (sometimes `send` will call itself again), we don't need to get the response for validation anymore
+                    res.send = responseSender;
+                    responseSender.call(res, isJson ? JSON.stringify(body): body);
```

Comment:
there should be no need to stringify an object to send it
was the thinking just to put back into `send` exactly the same format of thing we intercepted?

that's not a bad thought, but it is also making me think that we should take more care at the beginning, including:
- handling non-object and non-string parameters
- catching errors thrown by that original `JSON.parse`